Solutions/Armis/Data Connectors/ArmisAlerts/ArmisAlertSentinelConnector/__init__.py

```diff
@@ -0,0 +1,446 @@
+        encoded_hash = base64.b64encode(
+            hmac.new(decoded_key, bytes_to_hash, digestmod=hashlib.sha256).digest()
+        ).decode()
+        authorization = "SharedKey {}:{}".format(customer_id, encoded_hash)
+        return authorization
+
+    # Build and send a request to the POST API
+    def post_data(self, customer_id, body, log_type):
+        """Build and send a request to the POST API."""
+        method = "POST"
+        content_type = "application/json"
+        resource = "/api/logs"
+        rfc1123date = datetime.datetime.utcnow().strftime("%a, %d %b %Y %H:%M:%S GMT")
+        content_length = len(body)
+        timestamp_date = 'armis_alert_time'
+        try:
+            signature = self.build_signature(
+                rfc1123date,
+                content_length,
+                method,
+                content_type,
+                resource,
+            )
+        except ArmisException as err:
+            logging.error("Armis Alert Connector: Error occured: {}".format(err))
+            raise ArmisException(
+                "Armis Alert Connector: Error while generating signature for log analytics."
+            )
+
+        uri = (
+            "https://"
+            + customer_id
+            + ".ods.opinsights.azure.com"
+            + resource
+            + "?api-version=2016-04-01"
+        )
+
+        headers = {
+            "content-type": content_type,
+            "Authorization": signature,
+            "Log-Type": log_type,
+            "x-ms-date": rfc1123date,
+            "time-generated-field": timestamp_date,
+        }
+        try:
+            response = requests.post(uri, data=body, headers=headers)
+            if response.status_code >= 200 and response.status_code <= 299:
+                logging.info(
+                    "Armis Alert Connector: Data posted successfully to microsoft sentinel."
+                )
+            else:
+                raise ArmisException(
+                    "Armis Alert Connector: Response code: {} from posting data to log analytics.".format(
+                        response.status_code
+                    )
+                )
+
+        except ArmisException as err:
+            logging.error(err)
+            raise ArmisException(
+                "Armis Alert Connector: Error while posting data to microsoft sentinel."
+            )
+
+
+def main(mytimer: func.TimerRequest) -> None:
+    """
+    Start the execution.
+
+    Args:
+        mytimer (func.TimerRequest): This variable will be used to trigger the function.
+
+    """
+    utc_timestamp = (
+        datetime.datetime.utcnow().replace(tzinfo=datetime.timezone.utc).isoformat()
+    )
+    logging.info(
+        "Armis Alert Connector: Python timer trigger function ran at %s",
+        utc_timestamp,
+    )
+
+    armis_obj = ArmisAlert()
+    try:
+        armis_obj.check_data_exists_or_not_alert()
+    except ArmisDataNotFoundException:
```

Comment:
If no alert data found from Armis then now data found custom exception is raised from the child method to get the actual function app trigger result as success in the FunctionApp -> Monitor instead of fail.